caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -133,14 +140,19 @@ func (st ServerType) Setup(originalServerBlocks []caddyfile.ServerBlock,
 				return nil, warnings, fmt.Errorf("%s:%d: unrecognized directive: %s", tkn.File, tkn.Line, dir)
 			}
 
-			results, err := dirFunc(Helper{
+			h := Helper{
 				Dispenser:    caddyfile.NewDispenser(segment),
 				options:      options,
 				warnings:     &warnings,
 				matcherDefs:  matcherDefs,
 				parentBlock:  sb.block,
 				groupCounter: gc,
-			})
+				state:        make(map[string]interface{}),
+			}
+			h.state["tlsCertTags"] = tlsCertTags
```

Comment:
Again, initialize this where it is used; can remove this line

Reply:
Moved the initialisation of `state`